src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,201 @@
+package duke;
+
+
+
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.VBox;
+
+
+public class Duke {
+    private static boolean active;
+    private static TaskList taskList = new TaskList();
+    private static int listIndex = 0;
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/IMG_5401.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/IMG_4596.png"));
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    public String getResponse(String input) {
+        taskList = new TaskList(Storage.loadData());
+        return Parser.getResponse(input);
+    }
+
+    /**
+     * Iteration 1:
+     * Creates a label with the specified text and adds it to the dialog container.
+     * @param text String containing text to add
+     * @return a label with the specified text that has word wrap enabled.
+     */
+    private Label getDialogLabel(String text) {
+        Label textToAdd = new Label(text);
+        textToAdd.setWrapText(true);
+
+        return textToAdd;
+    }
+
+    /**
+     * awaken() awakens duke.Duke and allows one to input commands to duke.Duke.
+     */
+    public static void awaken() {
+
+        //load previous list data here maybe
+
+        taskList = new TaskList(Storage.loadData());
+
+        Duke.active = true;
+
+        Ui.sendStartMessage();
+
+        java.util.Scanner scanner = new java.util.Scanner(System.in);
+        while (active) {
+            String input = scanner.nextLine();
+            boolean cont = Parser.interpretCommand(input);
+            if (!cont) {
+                break;
+
+            }
+        }
+    }
+
+    /**
+     * deletes a task from the list that duke.Duke creates.
+     * @param taskToBeDeleted the index of the task to be deleted.
+     * @throws DukeException An exception stemming from incorrect or awkward input to duke.Duke.
+     */
+    public static String deleteTask(String taskToBeDeleted) throws DukeException {
```

Comment:
Perhaps you may want to consider defining the deleteTask method in the TaskList class (in order to maintain encapsulation and shortening the code for this class)?